thinc/backends/cupy_ops.py

```diff
@@ -170,39 +170,43 @@ def backprop_clipped_linear(
             )
 
     def backprop_hard_swish(self, dY, X, inplace: bool = False):
-        if X.dtype == "float32" and dY.dtype == "float32":
+        if X.dtype == dY.dtype and X.dtype in ("float32", "float64"):
             return _custom_kernels.backprop_hard_swish(dY, X, inplace=inplace)
         else:
             return super().backprop_hard_swish(dY, X, inplace=inplace)
 
     def backprop_hard_swish_mobilenet(self, dY, X, inplace: bool = False):
-        if X.dtype == "float32" and dY.dtype == "float32":
+        if X.dtype == dY.dtype and X.dtype in ("float32", "float64"):
             return _custom_kernels.backprop_hard_swish_mobilenet(dY, X, inplace=inplace)
         else:
             return super().backprop_hard_swish_mobilenet(dY, X, inplace=inplace)
 
     def mish(self, X, threshold=20.0, inplace=False):
-        if X.dtype == "float32":
+        if X.dtype in ("float32", "float64") and not inplace:
             return _custom_kernels.mish(X, inplace=inplace, threshold=threshold)
```

Comment:
Why do we need the `not inplace` check here?

Reply:
Fixed.